cli/src/main/java/de/jplag/cli/server/ResponseData.java

```diff
@@ -0,0 +1,56 @@
+package de.jplag.cli.server;
+
+import java.io.InputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Data for a http response
+ * @param stream The stream containing the binary data
+ * @param contentType The type of data
+ * @param size The total size of the data
+ */
+public record ResponseData(InputStream stream, ContentType contentType, int size) {
+
+    private static final Logger logger = LoggerFactory.getLogger(ResponseData.class);
+
+    /**
+     * Constructor with unknown type and size. Type will be set to PLAIN.
+     * @param data The binary data to respond with
+     */
+    public ResponseData(InputStream data) {
+        this(data, ContentType.PLAIN, 0);
+    }
+
+    /**
+     * Constructor with unknown size
+     * @param data The binary data
+     * @param contentType The type of content
+     */
+    public ResponseData(InputStream data, ContentType contentType) {
+        this(data, contentType, 0);
+    }
+
+    /**
+     * Creates a new instance for a given resource url.
+     * @param url The resource url
+     * @return The new response data
+     */
+    public static ResponseData fromResourceUrl(String url) {
+        if (url.endsWith("/")) {
+            return null;
+        }
+
+        InputStream inputStream = ResponseData.class.getResourceAsStream(url);
+
+        if (inputStream != null) {
+            return new ResponseData(inputStream, ContentType.fromPath(url));
+        } else {
+            logger.info(
+                    "No response data available for resource url {}. This is probably normal behaviour and the data will be pulled from a different source.",
```

Comment:
I'm not happy with this logging message as it too visible in a normal run and does not really help the end user (also: `this is probably normal behavior` lacks precision. If it ain't normal behavior, log it, otherwise, don't). Also, the important message `Press Enter key to exit...` gets easy to overlook when this is printed multiple times.